src/pages/index.js

```diff
@@ -119,7 +198,26 @@ profileAddButton.addEventListener("click", () => {
   profileAddPopup.open();
 });
 
-// // Initial render
-cardSection.renderItems();
+profileAvatarButton.addEventListener("click", () => {
+  profileAvatarPopup.open();
+});
 
-// Will add needs corrected in future
+document.addEventListener("DOMContentLoaded", async () => {
+  console.log("DOM fully loaded and parsed");
+  try {
+    const userInfoData = await api.getUserInfo();
+    currentUserId = userInfoData._id;
+    userInfo.setUserInfo(userInfoData);
+
+    const initialCardsData = await api.getInitialCards();
+    initialCardsData.forEach((data) => {
+      const cardElement = createCard(data, currentUserId);
+      cardSection.addItem(cardElement);
+    });
+  } catch (err) {
+    console.error("Error during initial load:", err);
+  }
+});
+
+// Initial render
+cardSection.renderItems();
```

Comment:
`cardSection.renderItems();` is not needed here because it's an empty call

It should work like that in line 213
```js
cardSection.renderItems(initialCardsData);

```